api/external/envoy/api/v2/core/base.proto

```diff
@@ -180,10 +180,22 @@ message HeaderValue {
   string value = 2 [(validate.rules).string = {max_bytes: 16384}];
 }
 
+// Header secret lookup key.
+message HeaderSecretValue { 
+    string name = 1;
+    string namespace = 2;
+}
+
 // Header name/value pair plus option to control append behavior.
 message HeaderValueOption {
   // Header name/value pair that this option applies to.
-  HeaderValue header = 1 [(validate.rules).message = {required: true}];
+  oneof header_option {
+    option (validate.required) = true;
+    // Explicitly specified herader
+    HeaderValue header = 1;
+    // Reference to header contained in a secret
```

Comment:
let's include some docs here that explain how the secret will get used (key/values in secret becoming the header key/values)

Reply:
I don't think that's appropriate here? solo-kit isn't responsible for how these get used downstream.

Reply:
These docs get rendered into the gloo docs https://docs.solo.io/gloo/latest/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/options/headers/headers.proto.sk/#headervalueoption

and no other project uses solo-kit (or currently plans to) beyond Gloo, so I think a bit more detail is merited here. 

i would propose a bit more detail, but not including anything gloo-specific (avoiding the word gloo/any of its implementation details, e.g. translation loop), but instead focus on the secret format and how it will get parsed. thoughts

Reply:
In principle, we could radically change how we implement this gloo-side and it would occur to no one to update this doc. This secret ref is totally independent of the secret shape. And the only secret with the correct shape on the gloo side is the HeaderSecret; referencing that here seems like it breaks the "don't reference gloo" principle.